### ISSUE TITLE
stability of _type-field

### ISSUE THREAD

**Issue by lindem** [offirgolan/ember-cp-validations#95]
I want to check for the type of a validator for a given attribute (to determine if there is a 'presence'-validation, to adapt css classes in a form for a given field, like, 'must be filled in'). Is it safe to use the `_type` field of a given validator for this purpose? The underscore hints at privateness or instability.

How likely is this to change? Is there a better way to check?

**Comment by lindem** [comment]
It seems that #93 is related to this issue as well; maybe there is will be a common solution for both issues.

**Comment by offirgolan** [comment]
You are welcome to use the `_type` field. I dont think it will be changing anytime soon but I will try to create a solution of #93 as soon as I have some time to work on it :smile:

**Comment by offirgolan** [comment]
#93 has been resolved and is in `v2.2.0`

**Comment by lindem** [comment]
Sorry for commenting on this again, but I have a pretty similar question once again: what about getting a validator function to validate arbitrary data from a model? To call the function I can do (from a live example, ember578 is the guid for a model):

```js
Ember.get(model, `validations._validators.${key}.${attr}[index].validate(theData, theOptions)`
```

Getting theOptions from the solution in #96. The actual questions are "How stable is the `_validators` field" and "is there an easier way". 

The need to do this arises from form validation (and I don't want the form to operate on the actual, validation-possessing object at first, at least not on all fields).

Should I open another issue for this?

**Comment by offirgolan** [comment]
I definitely think that using a container lookup is better than using the `_validators` object (thats what we do internally). I was hoping you can explain your use case a little better because Im not sure I follow

**Comment by lindem** [comment]
To further elaborate my use case: We have form fields which get told which model and attribute of that model they are going to fill in, and we want these form fields to fetch the validations from the model they are configured on (either we create the record on form display or we fetch an existing one). That all works fine so far. 

But. If you don't want to directly two-way-bind to a model attribute, because the user fills in a format different from the one stored (say, a date in German is dd.mm.yyyy instead of iso yyyy-mm-dd, which we want to store), you need some kind of intermediate step to translate into the format and then look at the validations from the input format *and* the stored format (because the stored format might have additional constraints). 

In short, we want to be able to translate to the stored format and arbitrarily want to check the validity before writing to the model using exactly the configuration of the validator we defined on the model.

I admit that this might not be the best way to do this kind of thing, but then, I am slowly getting into that topic with ember and cp-val's (doing forms since early 2000s, and validation with its many edge cases has never been easy). I am open to suggestions though :-)
